src/Common/tests/System.Xml.XPath/CoreFunctionLibrary/StringFunctionsTests.cs

```diff
@@ -529,6 +529,34 @@ public static void StringFunctionsTest2436()
 
         /// <summary>
         /// Verify result.
+        /// normalize-space("") = ""
+        /// </summary>
+        [Fact]
+        public static void StringFunctionsTest2473()
```

Comment:
Please add some tests for both functions with unicode characters and some including surrogate pairs. Try to do collisions if possible. Run it before and after change and compare results.